packages/tangle/filter.go

```diff
@@ -104,55 +105,78 @@ type PowFilter struct {
 	worker     *pow.Worker
 	difficulty int
 
+	nodesMessages      map[identity.ID]*pow.MessagesWindow
+	nodesMessagesMutex sync.Mutex
+
 	mu             sync.Mutex
-	acceptCallback func([]byte, *peer.Peer)
-	rejectCallback func([]byte, error, *peer.Peer)
+	acceptCallback func(*Message, *peer.Peer)
+	rejectCallback func(*Message, error, *peer.Peer)
 }
 
 // NewPowFilter creates a new PoW bytes filter.
 func NewPowFilter(worker *pow.Worker, difficulty int) *PowFilter {
+	pow.BaseDifficulty = difficulty
```

Comment:
Remove this line, since `pow.BaseDifficulty` is already set in `plugins/pow/pow.go`.